### DISCUSSION TITLE
[Proposal] Allow static keyword before arbitrary expression, defining an anymous static field in the surrounding type

### DISCUSSION THREAD

**Discussion by Meowtimer** [dotnet/csharplang#7825] [Language Ideas] [answered]
This would be a generalization of static lambdas in some sense in that instead of just static anonymous delegate fields you could define static fields of an arbitrary type. Meant for cases where moving out an expression into a static field would help performance but would unnecessarily disconnect the code from its usage context. Maybe there wouldn't be a good name for it except for `staticField123`.

Bogus example using a parser combinator library such as [Sprache](https://github.com/sprache/Sprache)
```csharp
using MyParserCombinatorLibrary;

public static class MyParsers
{

	public static Parser<int> MyParserWhichCanHaveAGoodName =
		from x in static ComplexSubParserExpression(123, 321)
		from x in static AnotherSubExpression(123, 321)
		select x
	;

}
```

Somewhat related is the code fix suggested for [CA1861](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1861). Using this language feature it could propose something like this instead:

```csharp
// A method argument
string message = string.Join(" ", static ["Hello", "world!"]);
```

**Comment by HaloFour** [comment, marked as answer]
See: #832

Which suggests both `static` locals as well as `static` expressions.

**Comment by aksanti** [comment]


**Reply by yaakov-h** [reply]
This seems to achieve nothing and is lighting up my favourite LLM/GPT-detector. What are you trying to say by removing the suggested language feature from its own proposal?

**Reply by AndersKehlet** [reply]
Yeah, that's GPT if ever I saw it.

@aksanti whatever you think you're doing, could you please not?

**Reply by aksanti** [reply]
My apologies please 🙏
